srv/modules/runners/validate.py

```diff
@@ -706,7 +706,18 @@ def fqdn(self):
 
         self._set_pass_status('fqdn')
 
-    # pylint: disable=line-too-long
+    def domain(self):
+        """
+        Verify that the domain grain exists
+        """
+        for minion_id in self.grains.keys():
+            domain = self.grains[minion_id]['domain']
+            fqdn = self.grains[minion_id]['fqdn']
+            msg = f"The domain grain is not set. The reason might be a missconfigured FQDN. Yours: <{fqdn}>"
```

Comment:
Just one 's' in misconfigured.  Did you want "Yours" to actually be minion_id.  If the fqdn is blank, they would still need to track it down.

Reply:
Could you please add here a message to the user, how user could verify if they have FQDN configured correctly from the Salt point of view.

Reply:
What about:

```
In order to determine a `domain` which is required for some features in DeepSea you need to set-up the FQDN properly.
Example:
A device with the hostname `myhost` in the parent domain example.com has the fully qualified domain name myhost.example.com. The FQDN uniquely distinguishes the device from any other hosts called myhost in other domains.
```

Reply:
@jschmid1 it does not say what exactly user should verify to make DeepSea happy, maybe include text like:
```Please make sure 'hostname --fqdn' returns FQDN, for example, host.example.com```?

Reply:
A message to the user would be quite nice.  I ran into this issue and had to dig around in deepsea to figure it out.

Reply:
```
In order to determine a `domain` which is required for some features in DeepSea you need to set-up the FQDN properly.
Example:
A device with the hostname `myhost` in the parent domain example.com has the fully qualified domain name myhost.example.com. The FQDN uniquely distinguishes the device from any other hosts called myhost in other domains.
Please make sure 'hostname --fqdn' returns FQDN, for example, host.example.com
Your currently configured FQDN is: <$fqdn>
```

Alright. Is this fine with everyone?

Reply:
sounds good.

Reply:
(sating hello sitting on vacation) Looks fine to me, just minor thing, since you're already using 'myhost' in the example and the example itself, probably it would be more appropriate:

> *Please make sure 'hostname --fqdn' returns **myhost.example.com** such as above*.

Reply:
ack